src/config.rs

```diff
@@ -2171,6 +2174,8 @@ impl Default for TiKvConfig {
 impl TiKvConfig {
     // TODO: change to validate(&self)
     pub fn validate(&mut self) -> Result<(), Box<dyn Error>> {
+        ensure_dir_exist(&self.storage.data_dir).unwrap();
```

Comment:
`ensure_dir_exist` is not intended to be called within `validate`, because we don't want to generate a directory after calling `validate`.